chrome/browser/android/adblock/adblock_bridge.cc

```diff
@@ -119,46 +119,36 @@ std::string escapeJavaScriptString(const std::string& subject) {
     U2029, "\\u2029");
 }
 
-std::string generateJavascript(
+const uint SELECTOR_GROUP_SIZE = 1024;
+#define DISPLAY_NONE_IMPORTANT " {display: none !important;}\n"
+
+std::string generateStylesheet(
   AdblockPlus::FilterEnginePtr filterEngine,
   std::string url,
   std::string domain) {
-  LOG(WARNING) << "Adblock: getting selectors for domain " << domain;
+  std::stringstream stylesheet;
   std::vector<std::string> selectors = filterEngine->GetElementHidingSelectors(domain);
-  LOG(WARNING) << "Adblock: got " << selectors.size() << " selectors for domain " << domain;
-
-  // build the string with selectors
-  std::stringstream ss;
-  ss << "var selectors = [\n";
-  for (unsigned int i=0; i<selectors.size(); i++) {
-    if (i > 0) {
-      ss << ", ";
+  
+  // selectors list should be splitted into groups of SELECTOR_GROUP_SIZE size:
+  // https://github.com/adblockplus/adblockpluschrome/blob/33991b8d88dae2712a9a3030d5143badbc155718/lib/contentFiltering.js#L46
+  
+  unsigned int last_group = 0;
```

Comment:
What about the following snippet?
There is no mess with the types, though it can seem that some variables are redundant.
It's impossible that the generated CSS is invalid if the number of selectors is divisible by the number of selectors in a group.
It should be easier to read it because the construction with if-for is pretty popular for joining elements of an iterable collection.
```C++
  uint32_t pendingSelectors = 0;
  auto ii_selectors = selectors.begin();
  if (ii_selectors != selectors.end())
  {
    stylesheet << *ii_selectors;
  }
  for (++pendingSelectors, ++ii_selectors;
       ii_selectors != selectors.end();
       ++pendingSelectors, ++ii_selectors)
  {
    if (pendingSelectors == selectorGroupSize) {
      stylesheet << displayNoneImportantCssRule;
      pendingSelectors = 0;
    } else {
      stylesheet << ", ";
    }
    stylesheet << *ii_selectors;
  }
  if (pendingSelectors > 0) {
    stylesheet << displayNoneImportantCssRule;
  }
```